### PR TITLE
Add column ordering to the aggregate report results view

### PR DESCRIPTION
**Pull request by wtritch**
This PR adds the ability to order Aggregate Report results table columns.

The order can be modified by selecting a column identifier and pressing the left/right buttons or by drag-and-drop re-ordering.

image

### CODE REVIEW

File: webapp/src/main/webapp/src/app/shared/order-selector/order-selector.component.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    this.itemsChange.emit(this.items);
+  }
+
+  public onDrag() {
+    this.selectedIdx = undefined;
+    this.selectedItem = undefined;
+  }
+
+  public onDrop() {
+    console.log("items", this.items);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/shared/order-selector/order-selector.component.ts]: remove?

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts [hunk 2]
```diff
@@ -175,6 +187,8 @@ export class AggregateReportComponent implements OnInit, OnDestroy {
       const item = this.itemMapper.map(this.assessmentDefinition, row, index);
       const subjectCode = row.assessment.subjectCode;
       const tableWrapper = tableWrappers.find(wrapper => wrapper.subjectCode == subjectCode);
+      //TODO this should come from the query
+      const columnOrder: string[] = ['organization', 'assessmentGrade', 'schoolYear', 'dimension'];
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: i take it we have to add it to the query all the way through the stack?

Comment (wtritch) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: yessir.  And display it on the sample table.  I'll do that in a follow on PR.

File: webapp/src/main/webapp/src/app/shared/order-selector/order-selector.component.ts (hunk 1 above)
Comment (wtritch) [reply webapp/src/main/webapp/src/app/shared/order-selector/order-selector.component.ts]: doh, nice catch

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts (hunk 2 above)
Comment (agorina) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: Just curious: are we allowing users to re-order *after* the report is generated? And if yes, are we planning on saving the changed order as part of their request?

Comment (wtritch) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: We are allowing the user to re-order after the report is generated.
Nothing they do while viewing the report results is "saved."  (Sorting, viewing the next page, displaying Percent or Numbers, etc including column ordering.)
We *should* allow the user to re-order the preview table much they can select whether to display Percent or Numbers.  We should persist their chosen order through the report generation and default their result table column order to the order the user chose at query time.

Comment (agorina) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: Thank you for clarifying it.

Comment (wtritch) [reply webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report.component.ts]: However, if the user exports a CSV of a table after modifying it, the CSV will respect the currently-displayed column order, sorting, and other display-time options.

**Pull request merged**